src/pytorch_lightning/strategies/fully_sharded_native.py

```diff
@@ -201,6 +205,26 @@ def _configure_launcher(self) -> None:
             self._launcher = _SubprocessScriptLauncher(self.cluster_environment, self.num_processes, self.num_nodes)
             self._rank_0_will_call_children_scripts = True
 
+    def _setup_model(self, model: torch.nn.Module) -> FullyShardedDataParallel:
+        """Wraps the model into a
+        :class:`~torch.distributed.fsdp.fully_sharded_data_parallel.FullyShardedDataParallel` module."""
+        # If model is already wrapped, we need to avoid sending the `auto_wrap_policy`
+        assert self.lightning_module is not None
+        if any(isinstance(mod, FullyShardedDataParallel) for _, mod in self.lightning_module.named_modules()):
+            if "auto_wrap_policy" in self.kwargs:
+                self.kwargs.pop("auto_wrap_policy")
```

Comment:
- Merge the ifs
- Names are unused
- we can safely del since you just checked it's inside

```suggestion
        if any(isinstance(mod, FullyShardedDataParallel) for mod in self.lightning_module.modules()) and "auto_wrap_policy" in self.kwargs:
            del kwargs["auto_wrap_policy"]
```